stablehlo/reference/Ops.cpp

```diff
@@ -1198,20 +1198,21 @@ Tensor allToAllOp(const Tensor &operand, Axis splitDimension,
   if (channelId > 0) processGroups = process->crossPartition(replicaGroups);
 
   auto processGroup = processGroups.findGroup(process->getId());
-  if (!processGroup)
+  if (!processGroup.has_value())
     llvm::report_fatal_error(invalidArgument(
         "Failed to find process group with process_id: (%d, %d)",
         process->getId().replicaId, process->getId().partitionId));
-  auto groupOperands = process->rendezvous(*processGroup, channelId, {operand})
-                           .getSortedTensors();
+  auto groupOperands =
+      process->rendezvous(processGroup.value(), channelId, {operand});
```

Comment:
nit:Can use `*processGroup`? here and below?

Reply:
any advantage of using *? `.value()` also returns reference to the object at std::optional and throws exception for null values, I think it is preferred  vs using * here?  also, it has better readability imo.

Reply:
sg, thanks for the review.